libres/lib/analysis/update.cpp

```diff
@@ -406,14 +379,16 @@ load_row_scaling_parameters(enkf_fs_type *target_fs,
                 }
             }
             const row_scaling_type *row_scaling =
-                local_dataset_get_row_scaling(dataset, key.c_str());
+                ministep->get_row_scaling(key);
+
             matrix_shrink_header(A, row_scaling->size(), matrix_get_columns(A));
             row_scaling_list.push_back(
                 std::pair<matrix_type *, const row_scaling_type *>(
                     matrix_alloc_copy(A), row_scaling));
         }
-        parameters[std::string(dataset_name)] = row_scaling_list;
+        parameters[ministep->name] = row_scaling_list;
```

Comment:
We only produce one "row_scaling_list" so we do not need to have the map of parameters to ministep names.